src/Zomp.SyncMethodGenerator/DiagnosticMessages.cs

```diff
@@ -26,5 +26,21 @@ internal static class DiagnosticMessages
         DiagnosticSeverity.Error,
         isEnabledByDefault: true);
 
+    internal static readonly DiagnosticDescriptor DuplicateUserMapping = new(
+        id: "ZSMGEN004",
+        title: "Duplicate user mapping",
+        messageFormat: "User mapping '{0}' is already defined",
+        category: Preprocessor,
+        DiagnosticSeverity.Error,
+        isEnabledByDefault: true);
+
+    internal static readonly DiagnosticDescriptor AttributeAndUserMappingConflict = new(
+        id: "ZSMGEN005",
+        title: "Attribute and user mapping conflict",
+        messageFormat: "Method '{0}' has both an attribute and a user mapping defined. The user mapping will be used.",
+        category: Preprocessor,
+        DiagnosticSeverity.Error,

```

Comment:
It's possible to have `[CreateSyncMethod]` and an user mapping on a method:
- The sync method will still be created. You can directly call it.
- However, whenever the async-to-sync rewriting happens, the user mapping will be used.

To prevent confusion why the method isn't being called, I've made this an error.